File: python/task2/S2_ARR_01.py
def clean_numbers(arr):
    result = []
    for x in arr:
        try:
            n = float(x)
            result.append(int(n) if n.is_integer() else n)
        except:
            pass
    return result

print(clean_numbers([" 1 ", "x", "2"]))

File: js/task2/S2_ARR_01.js
function cleanNumbers(arr) {
  return arr
    .map(v => Number(v.trim()))
    .filter(v => !Number.isNaN(v));
}

console.log(cleanNumbers([" 1 ", "x", "2", " 3.5 "])); // [1, 2, 3.5]
